app/controllers/api/v1/visualization/explore_controller.rb

```diff
@@ -5,10 +5,17 @@ module Visualization
       # the goal of this controller is to be business-logic free, and only amalgamate calls
       # to other controller/service methods when needed to save server round-trips
       class ExploreController < ApiBaseController
+        include Concerns::ApiCaching
         before_action :set_current_api_user!
+
         before_action :set_study
         before_action :check_study_view_permission
 
+        # don't cache cluster_options, as those are user-dependent.
+        before_action :check_api_cache!, except: :cluster_options
```

Comment:
I just found a corner case that we haven't considered - updating the study default options.  I changed both the default cluster/annotation, and I saw a message that said it was clearing the caches, and then re-caching the new "default" settings.  However, on page reload, the old defaults were still there, despite the values being changed in the study settings tab.  Looking at the logs, despite it saying that caches are cleared, the downstream viz requests still hit the cache, even if I comment out the line to pre-cache the defaults, so there is definitely a race condition somewhere.  I've tried ~10 times, and only once did it do the right thing.
```
Started PATCH "/single_cell/study/SCP13/snuc-seq/update_settings" for 127.0.0.1 at 2021-12-06 16:18:20 -0500
Processing by SiteController#update_study_settings as JS
  Parameters: {"authenticity_token"=>"[FILTERED]", "study"=>{"name"=>"sNuc-Seq", "embargo"=>"", "public"=>"1", "cell_count"=>"1402", "default_options"=>{"cluster"=>"DG", "annotation"=>"SUB-CLUSTER--group--study", "cluster_point_size"=>"6", "cluster_point_border"=>"false", "cluster_point_alpha"=>"1.0", "expression_label"=>"log(TPM)"}, "reviewer_access_attributes"=>{"expires_at"=>"2022-02-06"}}, "reviewer_access_actions"=>{"enable"=>"no", "reset"=>"no"}, "accession"=>"SCP13", "study_name"=>"snuc-seq"}
FireCloud API request (GET) https://api.firecloud.org/api/workspaces/single-cell-portal-development/snuc-seq-mongo-3-6/acl with tracking identifier: 6033f4a8e2413917a1d0ddcd
Disabling reviewer access in SCP13
  Rendering site/update_study_settings.js.erb
  Rendered site/_reviewer_access_fields.html.erb (Duration: 1.2ms | Allocations: 374)
  Rendered site/_study_settings_form.html.erb (Duration: 375.3ms | Allocations: 13942)
  Rendered site/_study_title_bar.html.erb (Duration: 0.4ms | Allocations: 49)
  Rendered layouts/_external_resources_view.html.erb (Duration: 0.8ms | Allocations: 144)
2021-12-06 16:18:28 -0500: Deleting caches for SCP13
2021-12-06 16:18:28 -0500: caches successfully cleared
  Rendered site/_study_description_view.html.erb (Duration: 83.1ms | Allocations: 2452)
  Rendered layouts/_download_link.html.erb (Duration: 1.4ms | Allocations: 347)
  Rendered layouts/_download_link.html.erb (Duration: 1.1ms | Allocations: 338)
  Rendered layouts/_download_link.html.erb (Duration: 1.4ms | Allocations: 336)
  Rendered layouts/_download_link.html.erb (Duration: 1.6ms | Allocations: 338)
  Rendered layouts/_download_link.html.erb (Duration: 1.7ms | Allocations: 338)
  Rendered layouts/_download_link.html.erb (Duration: 1.2ms | Allocations: 339)
  Rendered layouts/_download_link.html.erb (Duration: 1.0ms | Allocations: 338)
  Rendered layouts/_download_link.html.erb (Duration: 0.9ms | Allocations: 339)
  Rendered layouts/_download_link.html.erb (Duration: 1.0ms | Allocations: 338)
  Rendered layouts/_download_link.html.erb (Duration: 1.0ms | Allocations: 339)
  Rendered layouts/_download_link.html.erb (Duration: 0.9ms | Allocations: 338)
  Rendered site/_study_download_data.html.erb (Duration: 58.9ms | Allocations: 6534)
  Rendered site/update_study_settings.js.erb (Duration: 1167.9ms | Allocations: 49514)
Completed 200 OK in 7427ms (Views: 1169.0ms | MongoDB: 4.5ms | Allocations: 360440)


Started GET "/single_cell/api/v1/studies/SCP13/explore" for 127.0.0.1 at 2021-12-06 16:18:29 -0500
Processing by Api::V1::Visualization::ExploreController#show as JSON
  Parameters: {"study_id"=>"SCP13", "explore"=>{}}
Reading from API cache: _single_cell_api_v1_studies_SCP13_explore_
Filter chain halted as :check_api_cache! rendered or redirected
Completed 200 OK in 683ms (Views: 0.2ms | MongoDB: 0.3ms | Allocations: 39979)


Started GET "/single_cell/api/v1/studies/SCP13/clusters/_default?fields=coordinates%2Ccells%2Cannotation" for 127.0.0.1 at 2021-12-06 16:18:31 -0500
Started GET "/single_cell/api/v1/studies/SCP13/annotations" for 127.0.0.1 at 2021-12-06 16:18:31 -0500
Processing by Api::V1::Visualization::ClustersController#show as JSON
  Parameters: {"fields"=>"coordinates,cells,annotation", "study_id"=>"SCP13", "cluster_name"=>"_default", "cluster"=>{}}
Reading from API cache: _single_cell_api_v1_studies_SCP13_clusters__default_cluster_name__default_fields_coordinates_cells_annotation
Filter chain halted as :check_api_cache! rendered or redirected
Completed 200 OK in 456ms (Views: 0.2ms | MongoDB: 0.3ms | Allocations: 39968)
```